tests/unit/utils/test_validations.py

```diff
@@ -0,0 +1,498 @@
+    )
+
+    result = test_instance.validate()
+
+    assert len(result) == 1
+    assert (
+        result[0].message == "This file does not exist, check the following variable:"
+    )
+    assert result[0].property_name == "path_test_variable"
+
+
+def test_validate_file_not_a_path():
+    """
+    This test will validate if the class has been instanciated with a variable that was expecting a valid file path,
+    but it was provided with an invalid file path instead.
+    """
+    test_instance = TestClass(
+        path_test_variable="test",
+    )
+
+    result = test_instance.validate()
+
+    assert len(result) == 1
+    assert (
+        result[0].message
+        == "This is not a valid file path, check the following variable:"
+    )
+    assert result[0].property_name == "path_test_variable"
+
+
+def test_validate_datetime_correct():
+    """ """
+    test_instance = TestClass(
+        date_test_variable=datetime.today(),
+    )
+
+    result = test_instance.validate()
+
+    assert len(result) == 0
+
+
+def test_validate_date_correct():
+    """ """
+    test_instance = TestClass(
+        date_test_variable=date.today(),
+    )
+
+    result = test_instance.validate()
+
+    assert len(result) == 0
+
+
+def test_validate_date_incorrect():
+    """ """
+    test_instance = TestClass(
+        date_test_variable="test",
+    )
+
+    result = test_instance.validate()
+
+    assert len(result) == 1
+    assert (
+        result[0].message
+        == "This is not a valid date format, check the following variable:"
+    )
+    assert result[0].property_name == "date_test_variable"
+
+
+def test_validate_enum_correct():
+    test_instance = TestClass(test_enum_value=TestEnum.First_Value)
+
+    errors = test_instance.validate()
+
+    assert not any(errors)
+
+
+def test_validate_enum_incorrect():
+    test_instance = TestClass(test_enum_value=OtherTestEnum.First_Value)
+
+    errors = test_instance.validate()
+
+    assert any(errors)
+
+
+def test_validate_any_of_correct():
+    test_instance = TestClass(test_any_of_value="hi")
+
+    errors = test_instance.validate()
+
+    assert not any(errors)
+
+    test_instance = TestClass(test_any_of_value=3)
+
+    errors = test_instance.validate()
+
+    assert not any(errors)
+
+
+def test_validate_any_of_incorrect():
+    test_instance = TestClass(test_any_of_value=3.65)
+
+    errors = test_instance.validate()
+
+    assert any(errors)
+
+
+def test_validated_model_validation_correct():
+    test_instance = TestClass(
+        test_validated_model_class=OtherTestClass(str_test_variable_2="This is valid")
+    )
+
+    errors = test_instance.validate()
+
+    assert not any(errors)
+
+
+def test_validated_model_validation_incorrect():
+    test_instance = TestClass(
+        test_validated_model_class=OtherTestClass(str_test_variable_2=3.14)
+    )
+
+    errors = test_instance.validate()
+
+    assert any(errors)
+
+
+def test_validated_model_validation_incorrect_2():
```

Comment:
Can you give this a better name?

Reply:
Renamed to describe what it is testing a bit more